src/resolve/resolve-tool.c

```diff
@@ -430,7 +430,8 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         r = sd_bus_call(bus, req, DNS_CALL_TIMEOUT_USEC, &error, &reply);
         if (r < 0) {
-                log_error("%s: resolve call failed: %s", name, bus_error_message(&error, r));
+                if (warn_missing)
```

Comment:
wouldn't it be better to check for NXDOMAIN here explicitly? I mean, the bus call can fail for many reasons, but we should only handle the NXDOMAIN case here explicitly, no?

i.e. something like this:

`if (warn_missing || !sd_bus_error_has_name(&error, _BUS_ERROR_DNS "NXDOMAIN")) …`